frontend/public/module/k8s/resource.js

```diff
@@ -186,3 +188,45 @@ export const k8sWatch = (kind, query = {}, wsOptions = {}) => {
   wsOptions.path = path;
   return new WSFactory(path, wsOptions);
 };
+
+/**
+ * Use k8sWatch to wait for a resource to get into an expected condition.
+ * Watches for resource by kind, namespace and optional name.
+ * @param {K8sKind} kind
+ * @param {K8sResourceKind} resource
+ * @param {(kind: K8sResourceKind) => boolean} testFunction
+ * @param {number} timeoutInMs
+ * @returns {Promise<K8sResourceKind>}
+ */
+export const k8sWaitForUpdate = (kind, resource, testFunction, timeoutInMs) => {
+  const { namespace, name, resourceVersion } = resource.metadata;
+
+  if (testFunction(resource)) {
+    return Promise.resolve(resource);
+  }
+
+  const watcher = k8sWatch(kind, {
+    ns: namespace,
+    resourceVersion,
+  });
+  const closeConnection = () => watcher.destroy(5000);
```

Comment:
I do not want to mix the "destroy timeout" with the timeoutInMs parameter for "waiting until there is an update".

I set this to zero because we don't want to wait here on anything when closing the connection. Is this okay for you?

Reply:
I think you're misunderstanding what this value is... I don't think we need to pass anything. 

If you _wanted_ to pass something, you could pass the timeout value that was provided. Magic strings and numbers don't really have a lot of value imo. This value is just going to get echo'ed to anyone who is watching for the destroy event. Essentially that lifecycle of this `watcher` is your method here and thus no one outside of you are going to be listening to the "destroy" of the `watcher`.

Reply:
Remove the param from destroy, done